pkg/cache/types.go

```diff
@@ -0,0 +1,38 @@
+package cache
+
+import (
+	"errors"
+	"time"
+)
+
+var errWrongCacheType error = errors.New("requested cache type did not match previous cache with that name")
+var errTooCostly error = errors.New("item too costly for cache")
+var errTooFull error = errors.New("cache is full")
```

Comment:
If we are going to return these error lets export them so that assertions can be made outside of this package.